homeassistant/components/switchbot/switch.py

```diff
@@ -23,59 +38,103 @@
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Perform the setup for Switchbot devices."""
-    name = config.get(CONF_NAME)
-    mac_addr = config[CONF_MAC]
-    password = config.get(CONF_PASSWORD)
-    add_entities([SwitchBot(mac_addr, name, password)])
+async def async_setup_platform(
+    hass, config, async_add_entities, discovery_info=None
+) -> None:
+    """Import yaml config and initiates config flow for Switchbot devices."""
+
+    # Check if entry config exists and skips import if it does.
+    if hass.config_entries.async_entries(DOMAIN):
+        return
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_NAME: config[CONF_NAME],
+                CONF_PASSWORD: config.get(CONF_PASSWORD, None),
+                CONF_MAC: config[CONF_MAC].replace("-", ":").lower(),
+                CONF_SENSOR_TYPE: ATTR_BOT,
+            },
+        )
+    )
+
+
+async def async_setup_entry(hass, entry, async_add_entities) -> None:
+    """Set up Switchbot based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id][DATA_COORDINATOR]
+
+    bot_device = []
+
+    if entry.data[CONF_SENSOR_TYPE] == ATTR_BOT:
+        for idx in coordinator.data:
+            if idx == entry.unique_id.lower():
+
+                bot_device.append(
+                    SwitchBot(
+                        coordinator,
+                        idx,
+                        entry.data[CONF_MAC],
+                        entry.data[CONF_NAME],
+                        entry.data.get(CONF_PASSWORD, None),
+                        entry.options[CONF_RETRY_COUNT],
+                    )
+                )
+
+    async_add_entities(bot_device)
 
 
-class SwitchBot(SwitchEntity, RestoreEntity):
+class SwitchBot(CoordinatorEntity, SwitchEntity):
```

Comment:
Why do you remove `RestoreEntity`?

Reply:
Readded RestoreEntity. I though this was only needed for the "assumed switch state". It doesn't seem to have any effect.